lib/filterx/object-datetime.c

```diff
@@ -282,6 +282,8 @@ _strptime_eval(FilterXExpr *s)
         break;
     }
 
+  wall_clock_time_guess_missing_fields(&wct);
+
```

Comment:
It's unnecessary to guess the missing fields if we will return null because of a parsing error. This could go to the `if (end) { ... }` block below.